src/main/resources/assets/tconstruct/book/puny_smelting/en_us/upgrades/ranged/necrotic.json

```diff
@@ -0,0 +1,13 @@
+{
+  "modifier_id": "tconstruct:necrotic",
+  "text": [
+    {
+      "text": "Harvest the life of your foes in order to survive longer."
+    }
+  ],
+  "effects": [
+    "Critical hits steal life from your enemies to heal you",
+    "Multiple levels, increases chance of life steal per level",
```

Comment:
Its not chance based, its amount of life steal. If the melee section says its chance based it needs an update